File: packages/core/ios/Sources/DdSdk.m
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-Present Datadog, Inc.
 */

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(DdSdk, NSObject)

RCT_EXTERN_METHOD(initialize:(NSDictionary)configuration
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(setAttributes:(NSDictionary)attributes
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(setUser:(NSDictionary)user
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(setTrackingConsent:(NSString)trackingConsent
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(telemetryDebug:(NSString)message
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(telemetryError:(NSString)message
                 withStack:(NSString?)stack
                 withKing:(NSString?)kind
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: packages/core/src/sdk/DdSdk.ts
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-Present Datadog, Inc.
 */
import { getGlobalInstance } from '../utils/singletonUtils';

import { DdSdkWrapper } from './DdSdkInternal';
import type { DdSdkType } from './DdSdkInternal';

const CORE_MODULE = 'com.datadog.reactnative.core';
export const DdSdk = getGlobalInstance(
    CORE_MODULE,
    () => new DdSdkWrapper()
) as DdSdkType;
